### PR TITLE
Add benchmark for network with 4 peers

### PR DESCRIPTION
**Pull request by i1i1**
Signed-off-by: i1i1 <vanyarybin1@live.ru>

### CODE REVIEW

**Comment by eadventurous** [comment]
@i1i1 Should we close this one?

**Comment by i1i1** [comment]
@eadventurous i would wait for iroha_p2p merge. It should work fine at that time

File: iroha_client/benches/torii.rs [hunk 1]
```diff
@@ -12,20 +12,22 @@ use iroha_client::{
     client::{asset, Client},
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:12]: I think this should not be in benches `torii.rs` as it is a separate test case to check whole iroha and its TPS for 4 peers.

File: iroha_client/benches/torii.rs [hunk 2]
```diff
@@ -36,49 +38,67 @@ fn query_requests(criterion: &mut Criterion) {
...
+        asset_definition_id,
+    )));
+    vec![
+        create_domain.into(),
+        create_asset.into(),
+        create_account.into(),
+    ]
+}
+
+fn setup_bench() -> (Runtime, Client) {
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:67]:
```suggestion
fn setup_peer() -> (Runtime, Client) {
```

File: iroha_client/benches/torii.rs [hunk 3]
```diff
@@ -36,49 +38,67 @@ fn query_requests(criterion: &mut Criterion) {
...
-    let mut failures_count = 0;
+    let mut client = Client::new(&client_config);
+    client
+        .submit_all(setup_network())
+        .expect("Failed to create role.");
+    thread::sleep(std::time::Duration::from_millis(500));
+    (rt, client)
+}
+
+async fn setup_bench_network(n_peers: u32, max_txs_in_block: u32) -> Client {
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:82]:
```suggestion
async fn setup_n_peers(n_peers: u32, max_txs_in_block: u32) -> Client {
```

File: iroha_client/benches/torii.rs [hunk 4]
```diff
@@ -179,6 +185,81 @@ fn instruction_submits(criterion: &mut Criterion) {
...
+        success_count, failures_count
+    );
+    group.finish();
+
+    //if (failures_count + success_count) > 0 {
+    //    assert!(
+    //        success_count as f32 / (failures_count + success_count) as f32
+    //            > MINIMUM_SUCCESS_REQUEST_RATIO
+    //    );
+    //}
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:256]: We shouldn't have commented code

File: iroha_client/benches/torii.rs [hunk 5]
```diff
@@ -179,6 +185,81 @@ fn instruction_submits(criterion: &mut Criterion) {
...
+    let _ = group.bench_function("instructions", |b| {
+        b.iter(|| {
+            let ev = iroha_client
+                .listen_for_events(EventFilter::Pipeline(pipeline::EventFilter {
+                    entity: None,
+                    hash: None,
+                }))
+                .unwrap();
+
+            let quantity = txs_in_block / 100;
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:209]: Am I missing something or is this 0.5?

Comment (eadventurous) [reply iroha_client/benches/torii.rs:209]: Also have you tested it?

File: iroha_client/benches/torii.rs [hunk 6]
```diff
@@ -179,6 +185,81 @@ fn instruction_submits(criterion: &mut Criterion) {
...
+            failures_count += 100 - n;
+        })
+    });
+
+    println!(
+        "Success count: {}, Failures count: {}",
+        success_count, failures_count
+    );
+    group.finish();
+
```
Comment (eadventurous) [inline iroha_client/benches/torii.rs:250]: As our main aim is to calculate TPS then let's calculate and output it?

Comment (i1i1) [reply iroha_client/benches/torii.rs:250]: And see the error count

Comment (i1i1) [reply iroha_client/benches/torii.rs:250]: Like stress testing

File: iroha_client/benches/torii.rs (hunk 4 above)
Comment (i1i1) [reply iroha_client/benches/torii.rs:256]: It failed before. Success count was 0.2

**Comment by s8sato** [comment]
[Failed to run torii bench](https://github.com/s8sato/iroha/runs/3799133773?check_suite_focus=true#step:5:1496) by [my workflow](https://github.com/s8sato/iroha/blob/d6c65b6f8acbabc172c353520164567c88fa27e5/.github/workflows/_manual_bench.yml) on my fork.
Do you think there is something wrong with the workflow?

**Comment by s8sato** [comment]
Closed as #1952 inherits this

**Pull request closed without merging**